sdk/java/pulumi/src/test/java/com/pulumi/provider/internal/PropertyValueSerializerTest.java

```diff
@@ -413,6 +419,201 @@ void testDeserializingAssetAndArchiveWorks() {
         assertThat(archive).isEqualTo(fileArchive);
     }
 
+    class BasicResource  {
```

Comment:
Might be nice to have a test that serializes and then deserializes a resource, checking that we get the same thing back where possible?

Reply:
This isn't directly possible because we deserialize input Args classes but serialize resource classes. They aren't compatible - e.g. using Import vs Export decorators.